src/common/imagwebp.cpp

```diff
@@ -0,0 +1,231 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        src/common/imagwebp.cpp
+// Purpose:     wxImage WebP handler
+// Author:      Hermann Höhne
+// Copyright:   (c) Hermann Höhne
+// Licence:     wxWidgets licence
+/////////////////////////////////////////////////////////////////////////////
+
+// based on code by Sylvain Bougnoux, Khral Steelforge and Markus Juergens
+// see https://forums.wxwidgets.org/viewtopic.php?t=39212
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#if wxUSE_IMAGE && wxUSE_LIBWEBP
+
+#include "wx/imagwebp.h"
+#include "webp/demux.h"
+#include "webp/decode.h"
+#include "webp/encode.h"
+#include <vector>
+
+#ifndef WX_PRECOMP
+    #include "wx/intl.h"
+    #include "wx/log.h"
+#endif
+
+//-----------------------------------------------------------------------------
+// wxWEBPHandler
+//-----------------------------------------------------------------------------
+
+wxIMPLEMENT_DYNAMIC_CLASS(wxWEBPHandler, wxImageHandler);
+
+#if wxUSE_STREAMS
+
+#include <wx/mstream.h>
+#include <memory>
+#include <functional>
+
+typedef std::unique_ptr<WebPDemuxer, std::function<void(WebPDemuxer*)>> WebPDemuxerPtr;
```

Comment:
After looking at the code using this, I think it would be much more clear to have a wrapper class rather than using this unique pointer with a very custom deleter, i.e.

```
class wxWebPDemuxer
{
    WebPDemuxer* m_demuxer = nullptr;
    wxMemoryOutputStream m_mos;
    WebPData m_data;
};
```

Or, if possible, make the last 2 fields local variables in the function using them and then just use `unique_ptr<WebPDemuxer, WebPDemuxDelete>`.